### PR TITLE
NXP-32250: Admin Console warning on launch

### PR DESCRIPTION
**Pull request by Nishant0928**

### CODE REVIEW

File: src/app/shared/components/admin-warning/admin-warning.component.scss [hunk 1]
```diff
@@ -0,0 +1,55 @@
+.title {
+  font-family: "Inter";
```
Comment (swarnadipa-dev) [inline src/app/shared/components/admin-warning/admin-warning.component.scss:2]: Remove Inter font family

File: src/app/shared/components/admin-warning/admin-warning.component.ts [hunk 2]
```diff
@@ -1,7 +1,31 @@
-import { Component } from "@angular/core";
+import { Component, Input, HostListener } from "@angular/core";
```
Comment (swarnadipa-dev) [inline src/app/shared/components/admin-warning/admin-warning.component.ts:1]: Remove Hostlistener

File: src/app/shared/components/admin-warning/admin-warning.component.ts [hunk 3]
```diff
@@ -1,7 +1,31 @@
...
 })
-export class AdminWarningComponent {}
+export class AdminWarningComponent {
+  public doNotWarn: boolean = false;
+
+  constructor(private dialogService: MatDialog) {}
+
+  ngOnInit(): void {
+    const preference = localStorage.getItem("doNotWarn");
+    if (preference === "true") {
```
Comment (swarnadipa-dev) [inline src/app/shared/components/admin-warning/admin-warning.component.ts:16]: use !! operator

**Review by swarnadipa-dev — CHANGES_REQUESTED** [review]

File: src/app/admin-app.component.ts [hunk 4]
```diff
@@ -1,15 +1,20 @@
...
   selector: "admin-app-root",
   templateUrl: "./admin-app.component.html",
 })
 export class AdminAppComponent implements OnInit {
-  constructor(private store: Store) {}
+  constructor(private dialogService: MatDialog) {}
 
   ngOnInit(): void {
-    this.store.dispatch(authActions.getCurrentUser());
+    const preference = !!localStorage.getItem("doNotWarn");
```
Comment (swarnadipa-dev) [inline src/app/admin-app.component.ts:13]: use persistence service to get & set in localstorage

File: src/app/shared/components/admin-warning/admin-warning.component.scss [hunk 5]
```diff
@@ -0,0 +1,55 @@
+.title {
+  font-family: "Open Sans";
```
Comment (swarnadipa-dev) [inline src/app/shared/components/admin-warning/admin-warning.component.scss:2]: remove font-family property. It should take open sans as default font

File: src/app/shared/components/admin-warning/admin-warning.component.ts [hunk 6]
```diff
@@ -1,7 +1,29 @@
...
+  styleUrls: ["./admin-warning.component.scss"],
 })
-export class AdminWarningComponent {}
+export class AdminWarningComponent {
+  public doNotWarn: boolean = false;
+
+  constructor(private dialogService: MatDialog) {}
+
+  ngOnInit(): void {
+    const preference = localStorage.getItem("doNotWarn");
```
Comment (swarnadipa-dev) [inline src/app/shared/components/admin-warning/admin-warning.component.ts:15]: use persistence service

File: src/app/admin-app.component.ts [hunk 7]
```diff
@@ -1,15 +1,25 @@
...
 export class AdminAppComponent implements OnInit {
-  constructor(private store: Store) {}
+  constructor(
+    private dialogService: MatDialog,
+    private persistenceService: PersistenceService
+  ) {}
 
   ngOnInit(): void {
-    this.store.dispatch(authActions.getCurrentUser());
+    const preference = this.persistenceService.get("doNotWarn");
```
Comment (swarnadipa-dev) [inline src/app/admin-app.component.ts:17]: this variable is not required. We can directly write this as 
const doNotWarn = !!this.persistenceService.get("doNotWarn");;

File: src/app/admin-app.component.ts [hunk 8]
```diff
@@ -1,15 +1,21 @@
...
   templateUrl: "./admin-app.component.html",
 })
 export class AdminAppComponent implements OnInit {
-  constructor(private store: Store) {}
+  constructor( private dialogService: MatDialog) {}
 
   ngOnInit(): void {
-    this.store.dispatch(authActions.getCurrentUser());
+    const preference = localStorage.getItem("doNotWarn");
+    if (preference === "true") {
```
[the remark this answers is not in the document]
Comment (Nishant0928) [reply src/app/admin-app.component.ts]: done

File: src/app/app.module.ts [hunk 9]
```diff
@@ -23,13 +23,15 @@ import { authReducer } from "./auth/store/reducers";
 import { EffectsModule } from "@ngrx/effects";
 import * as authEffects from "./auth/store/effects";
 import { ngrxDevtools } from "../devtools/ngrx-devtools";
-import { ReactiveFormsModule } from "@angular/forms";
+import { FormsModule, ReactiveFormsModule } from "@angular/forms";
```
[the remark this answers is not in the document]
Comment (Nishant0928) [reply src/app/app.module.ts]: done

File: src/app/shared/components/admin-warning/admin-warning.component.scss (hunk 1 above)
Comment (Nishant0928) [reply src/app/shared/components/admin-warning/admin-warning.component.scss]: done

File: src/app/shared/components/admin-warning/admin-warning.component.ts (hunk 2 above)
Comment (Nishant0928) [reply src/app/shared/components/admin-warning/admin-warning.component.ts]: done

File: src/app/shared/components/admin-warning/admin-warning.component.ts (hunk 3 above)
Comment (Nishant0928) [reply src/app/shared/components/admin-warning/admin-warning.component.ts]: done

File: src/app/shared/components/admin-warning/admin-warning.component.ts [hunk 10]
```diff
@@ -1,7 +1,31 @@
...
+  }
+
+  onConfirm() {
+    if (this.doNotWarn) {
+      localStorage.setItem("doNotWarn", "true");
+    }
+    this.closeDialog();
+  }
+
+  closeDialog() {
```
[the remark this answers is not in the document]
Comment (Nishant0928) [reply src/app/shared/components/admin-warning/admin-warning.component.ts]: done

**Review by swarnadipa-dev — APPROVED** [review]

**Pull request closed without merging**